umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso19115_2/spatial.clj

```diff
@@ -81,17 +92,32 @@
     (into {} (for [[k ^String v] (partition 2 (str/split orbit-string #":? "))]
                [(keyword k) (Double/parseDouble v)]))))
 
+(defn- parse-vertical-domains
+  "Parse the vertical domain from the ISO XML document. Vertical domains are encoded in an ISO XML
+  document as a single string like this:
+  \"Type: Some type Value: Some value\""
+  [doc]
+  (when-let [vertical-domains (select doc vertical-string-xpath)]
+    (for [vertical-domain vertical-domains]
```

Comment:
I am defining what vertical-domains is and I want to test it to make sure it isn't nil before I go into the for loop. With your suggestion, I was assuming that in clojure this is done in the for loop with :let and :when.  Since I want to set the vertical-domains to a selection within the XML and then test it to see if it exists, changing the variable names doesn't help because I need to use the same name anyway.